pydisdrometer/plot/plot.py

```diff
@@ -32,22 +34,39 @@ def plot_dsd(dsd, range=None, log_scale=True, tighten=True):
     fig_handle: Figure Handle
 
     '''
+    ax =  parse_ax(ax)
+    fig = parse_ax(fig)
 
-    fig_handle = plt.figure()
+    if cmap is None:
+        colors = [('white')] + [(cm.jet(i)) for i in range(1, 256)]
+        cmap = mpl.colors.LinearSegmentedColormap.from_list(
+            'new_map', colors, N=256)
 
-    colors = [('white')] + [(cm.jet(i)) for i in range(1, 256)]
-    new_map = matplotlib.colors.LinearSegmentedColormap.from_list('new_map',
-                                                                colors, N=256)
+    if vmin is None:
+        vmin = np.nanmin(dsd.Nd)
+    if vmax is None:
+        vmax = np.nanmax(dsd.Nd)
 
-    plt.pcolor(dsd.time, dsd.diameter, np.log10(dsd.Nd.T), vmin=0.0,
-                figure=fig_handle, cmap=new_map)
+    if log_scale:
+        if vmin == 0.:
+            vmin = 0.1
+        norm = mpl.colors.LogNorm(vmin=vmin, vmax=vmax)
+    else:
+        norm = None
+    plt.pcolor(dsd.time, dsd.diameter, dsd.Nd.T, vmin=vmin, vmax=vmax,
+                figure=fig_handle, norm=norm, cmap=cmap)
```

Comment:
I get an error here that fig_handle has not been defined. I think you probably mean fig instead of fig_handle. Also I think if fig gets passed in as None, a figure won't be created, so we probably want to create a new figure on None.  I can make these changes if you agree.